### ISSUE TITLE
Significant drop in total storage 2.21b

### ISSUE THREAD

**Issue by false-9** [ClusterM/hakchi2#693]
![capture](https://user-images.githubusercontent.com/32708959/31443605-ccbd38fa-ae91-11e7-9706-680fb057aea4.PNG)

After synchronisation, the total storage indicator goes from 300MB to much less. Is this an inaccuracy? I am not using any mods, and only adding SNES games.

**Comment by zimmertr** [comment]
Many others have experienced this issue... Searching the issue board may be of use to you.

**Comment by hoderer** [comment]
I don't have this issue (SNESC germany), but have you checked savegames through the save state manager? And did it revert immediately when you use older version of hakchi? Is it SNESC or NESC? That's just a guess. For a better answer you should probably search a little bit in the forums or here like zimmertr said.

**Comment by false-9** [comment]
I did search other issues but couldn't see any pertaining to this, maybe I missed it,

**Comment by nerdnils** [comment]
First hit: https://github.com/ClusterM/hakchi2/issues/675

**Comment by false-9** [comment]
> First hit: #675

Sorry but I'm still confused, why is my space so small, especially compared to that guy who has lots more games than I have installed.

**Comment by Angelowaslike** [comment]
Same issue here. Tried countless times to uninstall and start from scratch but i still start at around 223 mb for storage.

**Comment by sebabadi** [comment]
Here 196,6 too, i read the faq, but numbers not clear: 290 total, 30 reserved, xx games, but remaining MB are not ok... where am i wrong?

**Comment by Drakonas** [comment]
@false-9 @sebabadi @Angelowaslike There are leftover files from older installs that cannot be removed manually. You need to do a complete reinstall of your SNES Classic. Backup your savestates with the savestate manager in Tools (the factory reset on the SNES Classic menu will erase all saves). Then, flash OG Kernel, then after that do an uninstall from the Kernel menu. Then, go to your SNES Classic and go to options and perform a factory reset. This is needed from 2.20 and older RC builds moving to 2.21b as the files have been moved around which are unremovable without a reset. After that just flash the custom kernel to reinstall Hakchi. With no mods installed you should see around 235 or so MB of max space available in Hakchi (the last of the two numbers). You can also free up more space by installing the "extra_space" mod if you feel like using an experimental feature. This will move all save states to an unused section of memory, freeing up more space for games. However this limits all savestate storage to 50MB, so pray you don't have more than that in size of all your Savestates.

**Comment by sebabadi** [comment]
@Drakonas hi, sorry but not very clear...
I did these things:
- deleted games (hey, how can i delete them? i delete the thick and i sicronyze?)
- deleted retroarch and company
- flashed original kernel
- reset console at tv
- hacked
- installed retroarch
- installed games

I don't have games savings right now, i have to start play from 0 and that's ok.

Maybe i did some error deleting games?

**Comment by Drakonas** [comment]
You need to follow my instructions in my post @sebabadi What you did won't help.

**Comment by sebabadi** [comment]
I can't undestand well your instructions...

**Comment by false-9** [comment]
> You need to do a complete factory reset of your SNES Classic. Backup your savestates with the savestate manager in Tools (the last step will erase all saves). Then, flash OG Kernel, then after that do an uninstall from the Kernel menu. Then, go to your SNES Classic and go to options and perform a factory reset. This is needed from 2.20 and older RC builds moving to 2.21b as the files have been moved around which are unremovable without a reset.


Cool, doing this now, thanks!

**Comment by Drakonas** [comment]
1. Backup your savestates with the savestate manager in Tools in the Hakchi program (Step 4 will erase all saves).
2. Flash OG Kernel.
3. Use the Uninstall function from the Kernel menu in Hakchi.
4. Go to the Options in the main menu on your SNES Classic and perform a factory reset.
5. Flash custom kernel.

@sebabadi You should see more space after this even when you install mods. (Remember, mods and savestates use up the max storage. Watch the number go down as you install mods and savestates)

**Comment by ajmi982** [comment]
Instructions are clear. Thanks. Will do this when I get home. Hopefully all my save states transfer because I was manually adding games via FPS prior to Hakchi support. 

Sent from my iPhone. Please pardon any spelling or grammar mistakes. 

On Oct 11, 2017, at 11:32 AM, sebabadi <notifications@github.com> wrote:

I can't undestand well your instructions...

**Comment by Drakonas** [comment]
@sebabadi Here are the steps:

1. Backup your savestates (Step 4 will erase all saves)
2. Flash OG Kernel.
3. Run the "Uninstall" function from the Hakchi "Kernel" menu.
4. Go to the main menu on your SNES Classic and go to "Options" and perform a factory reset.
5. Flash custom kernel.

You should see more space after this.

**Comment by nls44** [comment]
@Drakonas: I doubt this has anything to do with upgrading. I've just installed hakchi2 for the first time ever on my SNES Mini and this is what it's showing me:

![image](https://user-images.githubusercontent.com/1114597/31451240-e5ec6b2e-aeab-11e7-9345-323b4affac95.png)

That's with a grand total of 4 save states:

![image](https://user-images.githubusercontent.com/1114597/31451283-fd5a48a8-aeab-11e7-8111-539707747985.png)

**Comment by false-9** [comment]
Drakonas, I followed your steps and I'm looking at 238.2MB total storage before importing back my saves, after which I'm right back where I was, 196.6MB

This didn't do anything,

**Comment by Drakonas** [comment]
@bwq Your space is showing correctly. I don't see a problem. The SNES
Classic uses some of the storage for the original games, which are unable
to be removed. They can only be hidden by unchecking them in Hakchi.

**Comment by sebabadi** [comment]
???????

**Comment by dem1980** [comment]
Ok same for me.
Near 236 mo after reset snes mini (to delete savestates), uninstall hakchi2 and install custom kernel.
Then i add common plugin with default selected (retroarch with cores) and i have 199 mo.
I installed first version of hakchi2 when it was RC, i hope there are not old files (roms, settings,...) or folders that remain from this old build...

**Comment by nls44** [comment]
I think this has to do with Clover reserving 30 MB extra space instead of 10 MB like in previous releases. See here: https://github.com/ClusterM/hakchi2/issues/671

**Comment by K4Z1** [comment]
238,2 MB here too, after doing these steps. I got the same result before where I uninstalled first, then flashed the original Kernel, so I guess both works. 
- So after step 5. custom kernel flash, with original games: 238,2 MB
- After installing retroarch, snes9x core, music mod & extra space: 238,2 MB.
- After importing my saves (~51 MB) and synchronizing my ~70 games: 95,2/238,2 MB 
That is with extra-space mod installed - alright, that's an improvement! 
checking... saves work. Everything's fine :) 
Although it says the partition for extra-space-saves is only 50MB... what if my saves are 60-70 MB?

**Comment by THE-PBV** [comment]
I have 108 snes games installed
Retroarch with all mods
WITHOUT using the extra-space mod
![hakchit](https://user-images.githubusercontent.com/26841716/31454272-35c63504-aeb5-11e7-8434-3f760111d6b0.png)

**Comment by Linkinworm** [comment]
Is there a way to see how much space downloaded boxart takes up? I was making my own and found a better way to export images to save about 50% with no quality loss, only makes up the space for maybe 2 or 3 more games but before I convert a load of images again just wanted to know if just grabbing files from within hakchi is a better option

**Comment by ajmi982** [comment]
What number should we be seeing?

Sent from my iPhone. Please pardon any spelling or grammar mistakes. 

On Oct 11, 2017, at 11:48 AM, bwq <notifications@github.com> wrote:

@Drakonas: I doubt this has to do with upgrading. I've just installed hakchi2 for the first time ever on my SNES Mini and this is what it's showing me:



That's with a grand total of 4 save states:

**Comment by K4Z1** [comment]
@THE-PBV  It depends on how many save states you have, as they are HUGE ...you can check how big your saves are in the save-state manager, I guess yours are ~5 MB? ...I've made many save-states for ~20 games and they're 51 MB total. This will be subtracted from your max. available space. 

That's how I understand it:
![hak30](https://user-images.githubusercontent.com/32643158/31457354-c96ce5f8-aebc-11e7-8dc8-578f4bb22181.png)

with extra space:
![hak31](https://user-images.githubusercontent.com/32643158/31457359-cd7752dc-aebc-11e7-954a-4202efb5e2e0.png)
(pics are just photoshopped)

**Comment by THE-PBV** [comment]
@K4Z1 No thats my space without save states.

And your photoshop is totaly wrong.
It is 78,2MB USED from the 238   
Not Free space

**Comment by THE-PBV** [comment]
After i reset my mini snes to factory settings.
installed the original kernel.
end did is all over again.
After installing the hakchi kernel my free space was 238,2 MB
After installing the retroarch mod with snes9x2010 and tumbnail mod. it`s 232.2 MB
And after i installed the More space mod I have 257,7 MB

Looks good to me.
![extra](https://user-images.githubusercontent.com/26841716/31461220-26066166-aec9-11e7-8197-5a160cd85147.png)

**Comment by K4Z1** [comment]
I can't reach 257 MB, no matter what I try... only 238,2 MB.
I checked in FTP and ...in "/usr/share/games" only my NEW roms are listed, which looks good, 
but in "/var/lib/hakchi/squashfs/usr/share/games" it still shows all original 21 games with their suspend points it seems... is that normal?

**Comment by Drakonas** [comment]
@K4Z1 You have to install the extra_space mod to move the savestates to a separate location. And mods use up the 257/238.2MB of space too which will drop the size available.

**Comment by K4Z1** [comment]
@Drakonas yeah I posted before, I already installed extra space:
> So after step 5. custom kernel flash, with original games: 238,2 MB
>- After installing retroarch, snes9x core, music mod & extra space: 238,2 MB.
>- After importing my saves (~51 MB) and synchronizing my ~70 games: 95,2/238,2 MB

(saves were moved to extra space, but the extra space was never added to the 238,2 MB that hakchi shows)

I just wanted to let hakchi2 "know" that there's ~258 MB, like in @THE-PBV 's case. Oh well, as long as it works it's fine... just curious how you can tell hakchi that there's 258 MB with the extra space mod.

**Comment by Drakonas** [comment]
@K4Z1 hmm.. perhaps you need to flash og kernel again before step 5? from what THE-PBV said, that's what they did. I haven't installed the extra_space mod yet as A: I don't need it yet, and B: I am not in want to use an experimental feature when we don't know what this 50MB partition is there for in the first place.

**Comment by THE-PBV** [comment]
**1. Backup your save-states with save-state manager in the TOOL TAB .**
**2. Do a factory Reset on your Snes Mini.** (maybe not needed)
**3. Go to the KERNEL TAB and click Uninstall on the bottom** (This delete all hakchi files and flashed back the original kernel Your mini is now in factory state)
**4.Flash the custom Kernel** (now you have 238.2MB)
**5. Install modules when you need them.** (after installing retroarch, snes9x2010,and thumbnail mod you have 232,6MB free space)
**6.If you want more free space install the Extra-space mod** nou you have 257,7 MB
And without Retroarch installed you have over the 260MB free space

**Comment by K4Z1** [comment]
I found the problem! The music mod. It took 20MB for some reason (because the track is duplicated?), and the original menu music is on the SNESC anyway, just like the original 21 games and everything that is original, it just doesn't interfere with the available ~260 MB... so yeah, now I've got 257,7 MB  too :)

**Comment by stevedawg85** [comment]
When I go to 'uninstall modules', everything is already unchecked. Would I just check it and proceed?

**Comment by K4Z1** [comment]
@stevedawg85 yes. uninstall modules > check to uninstall.

I'm uninstalling the extra space mod now... I guess it's full or something, I can't make suspend points any more.
Edit: backed up saves, uninstalled extra space mod, imported saves, and I can make suspend points again. I don't know why I couldn't save, my only theory is that the 50MB extra partition was full (although it kinda worked before when I had the music mod oO

**Comment by Drakonas** [comment]
@stevedawg85 @K4Z1 it only detects what mods are installed if the console is on and connected to Hakchi at the time of running the install or uninstall mod function.

**Comment by AndreSanctus** [comment]
I notice that Retroarch save states (inside Retroarch) takes space too. Where can i delete them? Didn't see an option there...

I noticed now the following:

I install a fba rom, i play it, i save state inside retroarch, i play more. If i try to load state in the same snes mini boot instance, the retroarch load state works (and the save logically occupies space).

The problem is, if i reboot the snes mini, the saves aren't there anymore (inside retroarch) but the hakchi2 shows lesser space available. So, the save states are there, but unrecognizable after a snes mini reboot.

Then:
- i uninstalled all the emulator modules and i got the same initial 238MB total.
- i then re-installed the emulator modules again and got  198,8MB.

**Comment by Drakonas** [comment]
@AndreSanctus what FBA core are you using? ClusterM's or KMFDManic's?

**Comment by AndreSanctus** [comment]
@Drakonas, I think it's ClusterM's v1. I didn't know there is another. If it's "better", where can i download it?

**Comment by Drakonas** [comment]
@AndreSanctus No, that's the one you want to use. What FBA rom are you attempting?

In all honesty I would make a separate issue for this. That is a different issue.

**Comment by AndreSanctus** [comment]
@Drakonas, i can open a new topic, but i think this issue is related, this is about the total snes mini storage. Every time we save an internal retroarch save state, if it remains there or not (after a snes mini restart) it takes snes mini storage anyway but we can't manage those save states (delete, take out). If we keep saving inside retroarch, in time, not even one game will fit in snes mini.

(the FBA roms are that i am using are FBA v029740)

**Comment by Drakonas** [comment]
@AndreSanctus This issue thread was not really an issue, but rather was people not properly updating their SNES Classics. Yours is an actual bug it seems, from the retroarch cores. That needs a separate issue. I don't think it will be noticed otherwise.

**Comment by AndreSanctus** [comment]
@Drakonas , ok! I will start it. Thanks!
